eskip/string.go

```diff
@@ -99,8 +100,18 @@ func (r *Route) predicateString(sortPredicates bool) string {
 		predicates = appendFmtEscape(predicates, `Method("%s")`, `"`, r.Method)
 	}
 
-	for k, v := range r.Headers {
-		predicates = appendFmtEscape(predicates, `Header("%s", "%s")`, `"`, k, v)
+	headerKeys := make([]string, 0, len(r.Headers))
+
+	for k := range r.Headers {
+		headerKeys = append(headerKeys, k)
+	}
+
+	if sortHeaderPredicateValues {
+		sort.SliceStable(headerKeys, func(i, j int) bool { return r.Headers[headerKeys[i]] < r.Headers[headerKeys[j]] })
```

Comment:
please add line breaks around function body

Reply:
Here we sort by header value but we should sort by header name instead.
There could be different headers matching the same value, e.g. `Header("X-Foo", "avalue") && Header("X-Bar", "avalue")` but the configuration with the same header matching different values (e.g. `Header("X-Foo", "value1") && Header("X-Foo", "value2")`) is unlikely as it evaluates to false.

Reply:
Here we sort by header value but we should sort by header name instead.
There could be different headers matching the same value, e.g. `Header("X-Foo", "avalue") && Header("X-Bar", "avalue")` but the configuration with the same header matching different values (e.g. `Header("X-Foo", "value1") && Header("X-Foo", "value2")`) is unlikely as it evaluates to false.